Promote "Retriable and non-retriable pod failures for Jobs" to Beta (#37242)
kubernetes/website: content/en/docs/concepts/workloads/controllers/job.md

pod.  In particular, it needs to handle temporary files, locks, incomplete output and the like
caused by previous runs.

By default, each pod failure is counted towards the `.spec.backoffLimit` limit,
see [pod backoff failure policy](#pod-backoff-failure-policy). However, you can

> customize this by setting the Job's [pod failure policy](#pod-failure-policy).

Reviewer: ```suggestion
customize pod failure limit by setting the Job's [pod failure policy](#pod-failure-policy).
```

Author: I think in this `this` means "handling of pod failures". Updated to make it clear.

> see [pod backoff failure policy](#pod-backoff-failure-policy). However, this
> behavior can be customized by using Job's [pod failure policy](#pod-failure-policy).

Reviewer 2: (nit)
```suggestion
see [pod backoff failure policy](#pod-backoff-failure-policy). However, you can
customize this by setting the Job's [pod failure policy](#pod-failure-policy).
```

### Pod failure policy {#pod-failure-policy}

{{< feature-state for_k8s_version="v1.26" state="beta" >}}

{{< note >}}
You can only configure a Pod failure policy for a Job if you have the
`JobPodFailurePolicy` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
enabled in your cluster. Additionally, it is recommended
to enable the `PodDisruptionConditions` feature gate in order to be able to detect and handle
Pod disruption conditions in the Pod failure policy (see also:
[Pod disruption conditions](/docs/concepts/workloads/pods/disruptions#pod-disruption-conditions)). Both feature gates are

> available in Kubernetes v1.25.

Reviewer 2: Also revise this line. If you can, pick wording that lets you put `{{< skew currentVersion >}}` in place of a hard coded version number.

Author: Applied
